plugin/manager.go

```diff
@@ -18,7 +18,7 @@ var ErrNotRegularFile = errors.New("not regular file")
 
 // Manager manages plugins installed on the system.
 type Manager interface {
-	Get(ctx context.Context, name string) (Plugin, error)
+	Get(ctx context.Context, name string, pluginConfig map[string]string) (Plugin, error)
```

Comment:
@priteshbandi I'm confused why `get-plugin-metadata` has an optional `pluginConfig` in the request (see [spec](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#plugin-metadata)).

Could you elaborate a bit?

Reply:
What's meaning of passing `pluginConfig` to `get-plugin-metadata`?

Reply:
What I can think about is passing `{"experimental":true}` or `{"support_old_contract":true}` to `get-plugin-metadata` so that the plugin can have different responses.

Are above scenarios valid?

Reply:
Yeah for example if a plugin use to have only `SIGNATURE_GENERATOR.RAW`capability now plugin owners want to launch `SIGNATURE_GENERATOR.ENVELOPE` capability but wants it to only enable when users explicitly passes the flag.

Reply:
Makes sense.